src/server/api/email.js

```diff
@@ -0,0 +1,94 @@
+import {log} from '../utils';
+import fetch from 'node-fetch';
+import path from 'path';
+import {Database} from 'sqlite3';
+import {verifyEmail} from '../notifications/mailgun';
+import {AES, enc as encoding} from 'crypto-js';
+import {load as loadYaml} from 'node-yaml-config';
+
+const _config = loadYaml(path.join(__dirname, '..', '..', '..', 'config.yaml'));
+
+const validateEmail = (email) => {
+  const base = (string) => new Buffer(string).toString('base64');
+  const auth = base(`api:${_config.notifications.mailgun.public_key}`);
+  const basic = `Basic ${auth}`;
+
+  return fetch(`https://api.mailgun.net/v3/address/validate?address=${email}`, {
+    method: 'get',
+    headers: {
+      Authorization: basic,
+    },
+  });
+};
+
+export const verify = (req, res) => {
+  const {hash} = req.params;
+  const email = AES.decrypt(hash, _config.secret).toString(encoding.Utf8);
+  if (email.length === 0) {
+    res.sendStatus(422);
+  } else {
+    const db = new Database('canaritus.db');
+    db.run(`UPDATE emails SET verified='true' WHERE email='${email}'`, (err) => {
+      if (err !== null) {
+        log('DB', `Error updating "verify" on ${email}`, err);
+      } else {
+        res.sendStatus(200);
+      }
+    });
+    db.close();
+  }
+};
+
+export const subscribe = (req, res) => {
+  const {email} = req.body;
+  if (!email) {
+    res.status(422).send('email is missing from the post');
+  } else {
+    validateEmail(email).then((reply) => {return reply.json();}).then((data) => {
```

Comment:
I want to use async/await here.. but I'm having a super hard time configuring it for the project..